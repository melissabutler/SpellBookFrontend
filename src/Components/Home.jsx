import React, { useContext } from "react";
import CurrentUserContext from "../currentUserContext";

import HomeLoggedIn from "./HomeLoggedIn";
import HomeNoLogin from "./HomeNoLogin";

// import "./Home.css"

const Home = () => {
    let currentUser = useContext(CurrentUserContext);

    let page = "";

    if(!currentUser){
        page = <HomeNoLogin />
    } else {
        page = <HomeLoggedIn user={currentUser}/>
    }

    return (
        <div className="Home">
            <div className="Home-content">
                <h1 className="Home-title">Dungeons and Dragons 5th Edition SpellBook</h1>
                {page}
            </div>
            
        </div>
    )
}

export default Home;