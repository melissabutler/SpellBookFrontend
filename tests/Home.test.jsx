import React, { useContext } from "react";
import { describe, it, expect, vi } from "vitest";
import { render } from "@testing-library/react";
import Home from "../src/Components/Home";

describe(" Home", function() {

    vi.mock('react', async (importOriginal) => {
        const react = await importOriginal('react');
        return {
            ...react,
            useContext: () => ({ currentUser: {
                username: 'u1',
                email: 'email@email.com',
                isAdmin: false,
                characters: [
                    {
                    char_class: "cleric",
                    char_name: "Beebo",
                    id: 1,
                    lvl: 3
                },
                ]
            }})
    }})


    it("renders without crashing", async function() {
        render(<Home />)
    })

    it("matches snapshot", function() {
        const { asFragment } = render(<Home />);
        expect(asFragment()).toMatchSnapshot();
    })



})
