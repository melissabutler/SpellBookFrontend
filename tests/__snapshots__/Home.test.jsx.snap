// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[` Home > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Home"
  >
    <div
      class="Home-content"
    >
      <h1
        class="Home-title"
      >
        Dungeons and Dragons 5th Edition SpellBook
      </h1>
      <div
        class="Home-login"
      >
        <h3>
          Welcome back, 
        </h3>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
